addons/mail/models/mail_thread.py

```diff
@@ -2249,6 +2249,26 @@ def _message_compute_parent_id(self, parent_id):
                 parent_id = current_ancestor.parent_id.id if current_ancestor.parent_id else parent_id
         return parent_id
 
+    def _message_compute_subject(self, record_names=None, model_description=None):
```

Comment:
What about receiving the message as parameter, and use message.record_name when available, instead of giving an already existing record_names that you update afterwards ?

Reply:
I wanted to but we also use this in mail_compose_message (inside get_record_data) where we don't have access to the message yet.

I could apply the function later (after the mails are created) if it's a big deal but it seemed better to just leave it as it is.

Reply:
Passing record_names seems ok then. Concernign model_description, somehow we have to pass it through several methods, could be great to find a way to clean that and have it "accessible" easily. Outside the scope of this task, just thinking of it (cached stuff to find ?).

Reply:
Well it's only really used to fill some template values and the subject so I don't think it can ever be pulled from anywhere cleanly from what I've seen.

One option would be to just pass it into msg_vals, conceptually it might make sense and we'd avoid passing it as single parameter too often.

Or, what seems to make more sense, maybe the 2/3 models that actually use this feature could have a reserved computed field with something like 'model_type_name' or 'sub_model_name', which would act like the 'display_name' but for the name of the model instead of the record. But maybe we shouldn't do that specifically because it's so unused... 

At least I couldn't find a nice solution ^^